be/src/exec/pipeline/exchange/local_exchange.cpp

```diff
@@ -338,6 +338,26 @@ Status PassthroughExchanger::accept(const ChunkPtr& chunk, const int32_t sink_dr
     return Status::OK();
 }
 
+Status ConnectorSinkPassthroughExchanger::accept(const ChunkPtr& chunk, const int32_t sink_driver_sequence) {
+    size_t sources_num = _source->get_sources().size();
+    if (sources_num == 1) {
+        _source->get_sources()[0]->add_chunk(chunk);
+    } else {
+        // Scale up writers when current buffer memory utilization is more than 50% of the maximum and data processed
+        // is greater than current writer count * connector_sink_scaling_min_size. This also mean that we won't scale
+        // local writers if the writing speed can cope up with incoming data. In another word, buffer utilization is
+        // below 50%.
+        if (_writer_count < sources_num && _memory_manager->is_half_full() &&
+            _data_processed > _writer_count * config::writer_scaling_min_size_mb * 1024 * 1024) {
+            _writer_count++;
+        }
+        _source->get_sources()[(_next_accept_source++) % _writer_count]->add_chunk(chunk);
+        _data_processed += chunk->bytes_usage();
```

Comment:
why not use `_data_processing` instead of `_data_processed`? If we have already processed many chunks and hit some performance issue in BE, this check will be bypassed

Reply:
I don't think we can get accurate processing data in local exchange. Furthermore, what we are willing to do is that when it comes to a big query(a large number of data is going to be written), scale up as more writers as DOP to utilize our parallel processing capability. We don't need to save the number of writers, so in my opinion `_data_processed` is OK.